k8s/base/deploy-worker.yaml

```diff
@@ -0,0 +1,59 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/yannh/kubernetes-json-schema/master/master-standalone/deployment-apps-v1.json
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: worker
+  labels:
+    app.kubernetes.io/component: worker
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: worker
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/component: worker
+    spec:
+      securityContext:
+        fsGroup: 1000
+      volumes:
+        - name: tmp
+          persistentVolumeClaim:
```

Comment:
You should be using a `ephemeral` volume. https://kubernetes.io/docs/concepts/storage/ephemeral-volumes/#generic-ephemeral-volumes

A PVC can only be mounted by a single Pod at any given point (especially the `ReadWriteOnce` ones), meaning you can't really scale this worker more than 1 replica.

An ephemeral volume is created for each Pod independently.

Reply:
Leave the `storageClassName` out to keep things generic. It should use the default one, if not specified. The name can vary depending on the cluster.

Reply:
Okay, I was using ephemeral emptyDir volumes before and it was having a permission denied error for writing to them. Looking it up I couldn't figure out why, but something related to us not being root users in the pod, but it can be fixed by setting the `medium: Memory` on the volume for whatever reason so I'll do that.

Reply:
I linked *generic* ephemeral volumes.

Reply:
Which is different form `emptyDir` and please don't use memory. Some us don't have the space :)